scripts/ASTImportTest.sh

```diff
@@ -1,105 +1,190 @@
 #!/usr/bin/env bash
+# ------------------------------------------------------------------------------
+# vim:ts=4:et
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) solidity contributors.
+# ------------------------------------------------------------------------------
+# Bash script to test the import/exports.
+# ast import/export tests:
+#   - first exporting a .sol file to JSON, then loading it into the compiler
+#     and exporting it again. The second JSON should be identical to the first.
+
+set -euo pipefail
 
-set -e
-
-# Bash script to test the ast-import option of the compiler by
-# first exporting a .sol file to JSON, then loading it into the compiler
-# and exporting it again. The second JSON should be identical to the first
 READLINK=readlink
 if [[ "$OSTYPE" == "darwin"* ]]; then
     READLINK=greadlink
 fi
 REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
 SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
-SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
-SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+SOLC="${SOLIDITY_BUILD_DIR}/solc/solc"
+SPLITSOURCES="${REPO_ROOT}/scripts/splitSources.py"
+
+# shellcheck source=scripts/common.sh
+source "${REPO_ROOT}/scripts/common.sh"
+
+function print_usage
+{
+    fail "Usage: ${0} ast [--exit-on-error]."
+}
+
+function print_used_commands
+{
+    local test_directory="$1"
+    local export_command="$2"
+    local import_command="$3"
+    printError "You can find the files used for this test here: ${test_directory}"
+    printError "Used commands for test:"
+    printError "# export"
+    echo "$ ${export_command}" >&2
+    printError "# import"
+    echo "$ ${import_command}" >&2
+}
+
+function print_stderr_stdout
+{
+    local error_message="$1"
+    local stderr_file="$2"
+    local stdout_file="$3"
+    printError "$error_message"
+    printError ""
+    printError "stderr:"
+    cat "$stderr_file" >&2
+    printError ""
+    printError "stdout:"
+    cat "$stdout_file" >&2
+}
+
+IMPORT_TEST_TYPE=
+EXIT_ON_ERROR=0
+for PARAM in "$@"
+do
+    case "$PARAM" in
+        ast) IMPORT_TEST_TYPE="ast" ;;
+        --exit-on-error) EXIT_ON_ERROR=1 ;;
+        *) print_usage ;;
```

Comment:
I think we should exit with an error if we print usage and it was not explicitly requested with something like `--help`. Otherwise it's too easy not to notice that you're giving it wrong arguments when calling it from another script.

Reply:
Added an additional error message here. `print_usage` still returns exit code `-1` here.

Reply:
Ah, I didn't even realize that because the name does not suggest to me at all that it would exit with an error.